src/engine/api/json_factory.cpp

```diff
@@ -233,6 +233,7 @@ util::json::Object makeRouteStep(guidance::RouteStep step, util::json::Value geo
     route_step.values["distance"] = std::round(step.distance * 10) / 10.;
     route_step.values["duration"] = std::round(step.duration * 10) / 10.;
     route_step.values["name"] = std::move(step.name);
+    route_step.values["ref"] = std::move(step.ref);
```

Comment:
@daniel-j-h The handling of `name` is probably fine though. In real life I don't really think it's optional for streets to not have names anymore, so an empty name field implies that we don't have the data, whereas a missing field implies it doesn't exist.